### PR TITLE
Added flushRX function to HardwareSerial

### PR DESCRIPTION
**Pull request by emsal0**
Snowbots used/will be using the `flushRX()` function in at least the GPS driver since last year. This flushes the RX buffer rather than the TX buffer (which `HardwareSerial::flush()` does) (don't quote me on this explicitly, fact-check with @Dulluhan  to be absolutely certain what this is for).

Previously, we would actually go to the install directory of the arduino package to include this functionality. This pull request (and indeed, this entire fork) was mostly created to have better documentation and reliability regarding that -- we'd still need to figure out exactly how to link this fork into our IGVC-2017 project (thinking Chef) but I feel that this is a step in the right direction.

### CODE REVIEW

**Comment by garethellis0** [comment]
Have you done some testing to ensure this is working correctly? Also, @Dulluhan , would you mind taking a look at this and making sure that:
- the comment above accurately describes what flushRX() is used for
- that it's implemented correctly? 

Also, great work Em, thanks!

**Comment by emsal0** [comment]
Need to figure out how to use a sketch with this custom build -- perhaps we could try it next meeting. Would also probably need to add something to the chef scripts to put it in the proper place upon running the script.

**Comment by garethellis0** [comment]
Ok, sounds good. As an aside, we moved off of chef to plain ol' bash, but just let me know what needs to get put where and I'll make sure the setup script does it.

**Comment by Dulluhan** [comment]
Realized I've written about this is in the past. [http://c4coffee.ca/2016/03/27/Arduino-Serial-Communication-Fix.html](http://c4coffee.ca/2016/03/27/Arduino-Serial-Communication-Fix.html) See if this is good information. 

I will look into making a repo solution for it.

**Comment by garethellis0** [comment]
As far as I can see, this solution is the exact same as the one on your blog @Dulluhan . What do you mean by a "repo solution"? This repo is under Snowbots already?

**Comment by Dulluhan** [comment]
An inline solution to our current code rather than adjusting things manually.

**Comment by emsal0** [comment]
Hmm, I think it could technically be possible to extend another class out of `HardwareSerial`. We're lucky that the methods we're trying to access are `protected` rather than `private`.

**Comment by Dulluhan** [comment]
I did some research, maybe we can try to just use serial.read() to clear the queue until an identifier byte is found. Then see if that will affect how we are performing. Maybe I was just dumb about things because apparently most people are getting problems from writing to serial. Arduino writes to serial faster than the output buffer is cleared. Hence why they changed the flush functionality after Arduino 1.0 

tl;dr.  Remove my dumb function and test with Serial.read(). If it works fine, hurray im dumb. If not. rip.
